analyses/cms-open-data-ttbar/analysis.py

```diff
@@ -1,518 +1,361 @@
+            btag[i] = std::max({jet_btag[j1], jet_btag[j2], jet_btag[j3]}) > 0.5;
+        }
+        return btag;
+        """,
+    )
+
+    # Evaluate mass of trijet with maximum pt and btag higher than threshold
+    df = df.Define(
+        "trijet_mass",
+        """
+        double mass{};
+        double Pt{};
+        double indx{};
+        for (int i = 0; i < ntrijet; ++i) {
+            if ((Pt < trijet_pt[i]) && (trijet_btag[i])) {
+                Pt = trijet_pt[i];
+                indx = i;
+            }
+        }
+        mass = trijet_p4[indx].M();
+        return mass;
+        """,
+    )
+
+    return df
+
+
+def book_histos(
+    df: ROOT.RDataFrame,
+    process: str,
+    variation: str,
+    nevents: int,
+) -> list[AGCResult]:
+    """Return the RDataFrame results pertaining to the desired process and variation."""
+    # Calculate normalization for MC
+    x_sec = XSEC_INFO[process]
+    lumi = 3378  # /pb
+    xsec_weight = x_sec * lumi / nevents
+    df = df.Define("weights", str(xsec_weight))  # default weights
+
+    if variation == "nominal":
+        # Jet_pt variations definition
+        # pt_scale_up() and pt_res_up(jet_pt) return scaling factors applying to jet_pt
+        # pt_scale_up() - jet energy scaly systematic
+        # pt_res_up(jet_pt) - jet resolution systematic
+        df = df.Vary(
+            "jet_pt",
+            "ROOT::RVec<ROOT::RVecF>{jet_pt*pt_scale_up(), jet_pt*jet_pt_resolution(jet_pt.size())}",
+            ["pt_scale_up", "pt_res_up"],
+        )
+
+        if process == "wjets":
+            # Flat weight variation definition
+            df = df.Vary(
+                "weights",
+                "weights*flat_variation()",
+                [f"scale_var_{direction}" for direction in ["up", "down"]],
+            )
+
+    # Event selection - the core part of the algorithm applied for both regions
+    # Selecting events containing at least one lepton and four jets with pT > 25 GeV
+    # Applying requirement at least one of them must be b-tagged jet (see details in the specification)
+    df = (
+        df.Define("electron_pt_mask", "electron_pt>25")
+        .Define("muon_pt_mask", "muon_pt>25")
+        .Define("jet_pt_mask", "jet_pt>25")
+        .Filter("Sum(electron_pt_mask) + Sum(muon_pt_mask) == 1")
+        .Filter("Sum(jet_pt_mask) >= 4")
+        .Filter("Sum(jet_btag[jet_pt_mask]>=0.5)>=1")
+    )
+
+    # b-tagging variations for nominal samples
+    if variation == "nominal":
+        df = df.Vary(
+            "weights",
+            "ROOT::RVecD{weights*btag_weight_variation(jet_pt[jet_pt_mask])}",
+            [
+                f"{weight_name}_{direction}"
+                for weight_name in [f"btag_var_{i}" for i in range(4)]
+                for direction in ["up", "down"]
+            ],
+        )
+
+    # Define HT observable for the 4j1b region
+    # Only one b-tagged region required
+    # The observable is the total transvesre momentum
+    # fmt: off
```

Comment:
What is this switch doing?

Reply:
black doesn't format this block well, see https://github.com/psf/black/issues/67